zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -366,7 +366,12 @@ private void appendToQueuedBuffer(ByteBuf buf) {
     void processMessage(ByteBuf buf) {
         checkIsInEventLoop("processMessage");
         LOG.debug("0x{} queuedBuffer: {}", Long.toHexString(sessionId), queuedBuffer);
-
+        
+        if (closingChannel) {
+            LOG.info("Closing connection to {}", getRemoteSocketAddress());
+            return;
+        }
+        
```

Comment:
```suggestion

        if (closingChannel) {
            LOG.debug("Drop incoming message during connection closing for session 0x{}", Long.toHexString(sessionId));
            return;
        }

```

I am not sure `getRemoteSocketAddress` is safe after channel closed. Also, postmortem log probably don't deserve `LOG.info`.

Connection is closing, any postmortem